examples/vault-prod-ready/README.md

```diff
@@ -0,0 +1,180 @@
+
+This example icombines the followings: Vault auto unseal - Vault private cluster - Consul with encryption and also adds some stuff that we expect in a prod ready working state, mainly:
+* Secure instances using security groups and remove [0.0.0.0/0]
+* Enable ACL on consul and pass the token in a secure way using AWS Secrets manager
+* Enable encryption on Consul (based on https://github.com/hashicorp/terraform-aws-consul/tree/master/examples/example-with-encryption)
+* Auto generate gossip_encryption_key and store it in AWS Secrets manager
+* Only allow ssh from Vpn server
+* Creates an ELB in front of Consul instances
+* Creates an ELB in front of Vault instances
+* Removed the block for deploying cluster in default VPC and availability zones, this needs to be set explicitly to enforce security
+
+This example use Terragrunt to solve some dependencies but can by used with terraform also.
+
+Thanks @mkrull for the ACL token provision
+
+# HOWTO
+* 1 - Run terragrunt
+* 2 - run `vault operator init` on one of the vault servers
+* 3 - To login in consul look for master token under aws secrets manager 
+
+# Vault auto unseal example
+
+This folder shows an example of Terraform code that deploys a [Vault][vault] cluster
+in AWS with [auto unseal][auto_unseal]. Auto unseal is a Vault feature
+that automatically [unseals][seal] each node in the cluster at boot using [Amazon KMS][kms].
+Without auto unseal, Vault operators are expected to manually unseal each Vault node
+after it boots, a cumbersome process that typically requires multiple Vault operators
+to each enter a Vault master key shard.
+
+This example creates a private Vault cluster that is accessible only from within
+the VPC within the AWS account in which it resides, or other VPCs that are peered
+with the Vault VPC. The Vault cluster uses [Consul][consul] as the storage backend,
+so this example also deploys a separate Consul server cluster using the
+[consul-cluster module][consul_cluster] from the Consul AWS Module. Each of the
+servers in this example has [Dnsmasq][dnsmasq] installed (via the [install-dnsmasq module][dnsmasq_module])
+or [setup-systemd-resolved][setup_systemd_resolved] (in the case of Ubuntu 18.04)
+which allows them to use the Consul server cluster for service discovery and thereby
+access Vault via DNS using the domain name `vault.service.consul`.
+
+For more info on how the Vault cluster works, check out the [vault-cluster][vault_cluster]
+documentation.
+
+**Note**: To keep this example as simple to deploy and test as possible, it deploys
+the Vault cluster into your default VPC and default subnets, all of which are publicly
+accessible. This is OK for learning and experimenting, but for production usage,
+we strongly recommend deploying the Vault cluster into the private subnets of a custom VPC.
+
+**Billing Warning**: Every time you create a KMS key, you're charged $1 for the month,
```

Comment:
Because of this, all of our KMS tests take in a KMS key ID as an input variable. That way, we can create a single KMS key we use for testing, rather than a new one for each test run.

Reply:
Ah, looks like you may be doing just that. You may want to explain that here.